scio-core/src/main/scala/com/spotify/scio/values/SideMap.scala

```diff
@@ -18,4 +18,8 @@ package com.spotify.scio.values
 
 import scala.collection.mutable.{ArrayBuffer, Map => MMap}
 
-case class SideMap[K, V](side: SideInput[MMap[K, ArrayBuffer[V]]])
+@deprecated("Use SCollection[(K, V)]#asMultiMapSideInput instead", "0.8.0")
+case class SideMap[K, V](side: SideInput[MMap[K, ArrayBuffer[V]]]) {
+  private[values] def asImmutableSideInput: SideInput[Map[K, Iterable[V]]] =
+    side.map(_.mapValues(_.toIterable).toMap)
```

Comment:
```suggestion
    side.map(_.iterator.mapValues(_.toIterable).toMap)
```

I'm not 100% but this should avoid making an intermediate `Map` in `mapValues`.

Reply:
yup it;s avoided! In general case these days I'm pro removing mapValues and replace it with map(case) ...